rpcs3/rpcs3qt/log_frame.cpp

```diff
@@ -430,25 +430,26 @@ void log_frame::UpdateUI()
 {
 	const auto start = steady_clock::now();
 
+	const auto tty = g_fxo->get<tty_t>();
+
 	// Check TTY logs
-	while (const u64 size = std::max<s64>(0, g_tty_size.load() - (m_tty_file ? m_tty_file.pos() : 0)))
+	if (tty && tty->fd && tty->fd.size() && m_tty_size != tty->fd.size())
 	{
 		std::string buf;
-		buf.resize(size);
-		buf.resize(m_tty_file.read(&buf.front(), buf.size()));
-
-		if (buf.find_first_of('\0') != umax)
 		{
-			m_tty_file.seek(s64{0} - buf.size(), fs::seek_mode::seek_cur);
-			break;
+			std::lock_guard lock(tty->mtx);
+			const auto pos = tty->fd.pos();
+			tty->fd.seek(m_tty_size);
+			tty->fd.read(buf, tty->fd.size() - m_tty_size);
+			tty->fd.seek(pos);
+			m_tty_size = tty->fd.size();
 		}
```

Comment:
This part seems unwanted in the disabled state.
My suggestion is to move this scope (including the std::string buf) as nested scope behind the following m_TTYAct check.
And then add an else path for updating the m_tty_size

Reply:
this here was my main complaint though